xarray/tests/duckarrays/base/constructors.py

```diff
@@ -0,0 +1,59 @@
+import hypothesis.strategies as st
+import numpy as np
+import numpy.testing as npt
+from hypothesis import given, note, settings
+
+import xarray as xr
+
+from . import strategies
+
+
+class VariableConstructorTests:
+    def check(self, var, arr):
+        self.check_types(var, arr)
+        self.check_values(var, arr)
+        self.check_attributes(var, arr)
+
+    def check_types(self, var, arr):
+        # test type of wrapped array
+        assert isinstance(
+            var.data, type(arr)
+        ), f"found {type(var.data)}, expected {type(arr)}"
+
+    def check_attributes(self, var, arr):
+        # test ndarry attributes are exposed correctly
+        assert var.ndim == arr.ndim
+        assert var.shape == arr.shape
+        assert var.dtype == arr.dtype
+        assert var.size == arr.size
+        assert var.nbytes == arr.nbytes
+
+    def check_values(self, var, arr):
+        # test coersion to numpy
+        npt.assert_equal(var.to_numpy(), np.asarray(arr))
+
+    @staticmethod
+    def create(shape, dtypes):
+        return strategies.numpy_array(shape, dtypes)
+
+    @given(st.data())
+    @settings(deadline=None)
+    def test_construct(self, data):
+        arr = data.draw(
+            strategies.duckarray(lambda shape, dtypes: self.create(shape, dtypes))
+        )
+
+        # TODO generalize to N dimensions
+        dims = ["dim_0", "dim_1", "dim_2"]
```

Comment:
This is related to https://github.com/pydata/xarray/pull/4972#discussion_r941690041 because if dimension names aren't part of the strategies then we could just have a utility function which generates lists of dimension names of the form `dim_*`.

Reply:
Oh the exact function I just wrote already exists as `utils.create_dimension_names`, my bad! :sweat_smile: